src/odemis/acq/fastem.py

```diff
@@ -587,4 +590,11 @@ def _pass_future_progress(sub_f, start, end):
 
     if len(das) != 1:
         logging.warning("Expected 1 DataArray, but got %d: %r", len(das), das)
+
+    # Switch immersion mode back on, so we can do use the system normally.
+    stream.emitter.immersion.value = True
```

Comment:
This is a little odd/unclear to me. Typically, you want to configure the hardware with the right settings *before* you do an acquisition. After an acquisition, you don't need to change the setting. It's up to the *next* acquisition to do this work. (Here, the chances that the next acquisition is an another overview is actually quite high, so it's just extra work/extra chances to mess up with some other settings).

So my question is: what is the "normal use of the system"? Is that because for now the user still relies on the TFS GUI to do some tasks, and it's inconvenient to switch immersion mode in their software? Or is there some acquisition in Odemis which do no set the immersion mode properly?

Reply:
Yes, with the current state of the system after acquiring an overview image the change of acquiring another one directly is quite low. You will need to do SEM focusing through the TFS GUI first and indeed it's inconvenient to switch immersion mode in their software. I will update the comment to explain this a bit better.